client/src/components/Rate/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ *
+ * Rate
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import StyledRate from './StyledRate';
+
+import RatingItem from './RatingItem';
+/* eslint-disable react/no-array-index-key */
+
+const range = (min, max) =>
+  Array(max - min + 1)
+    .fill()
+    .map((_, i) => min + i);
+
+const Rating = ({ onChange, value, clickable }) => {
+  return (
+    <div className="rating">
+      {range(1, 5).map((item, index) => (
+        <RatingItem
+          colored={value >= item}
+          checked={value === item}
+          value={item}
+          onChange={onChange}
+          clickable={clickable}
+          htmlFor={`star_${index}`}
+          key={`star_${index}`}
+        />
+      ))}
+    </div>
+  );
+};
+
+function Rate({ clickable, value, size }) {
+  return (
+    <StyledRate
+      className={`${size} ${clickable ? 'rating-item-clickable' : ''}`}
+    >
+      <Rating value={value} clickable={clickable} />
+    </StyledRate>
+  );
+}
+
+Rate.defaultProps = {
+  value: 0,
+  clickable: false,
+  size: 'small'
+};
+
+Rate.propTypes = {
+  value: PropTypes.oneOf([0, 1, 2, 3, 4, 5, NaN]),
```

Comment:
why Nan ?

Reply:
I can't remember, but I think that at some point the API returned NaN which is not the case anymore. Maybe it should be removed.